src/utils/anonymousUserMerge.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  SHARED_PREF_EMAIL,
+  SHARED_PREF_USER_ID,
+  ENDPOINT_GET_USER_BY_USERID,
+  ENDPOINT_GET_USER_BY_EMAIL,
+  ENDPOINT_MERGE_USER
+} from 'src/constants';
+import { AnonymousUserEventManager } from './anonymousUserEventManager';
+import { baseIterableRequest } from '../request';
+import { IterableResponse } from '../types';
+
+export class AnonymousUserMerge {
+  private anonymousUserManager = new AnonymousUserEventManager();
+
+  mergeUserUsingUserId(destinationUserId: string): void {
+    const sourceUserId = localStorage.getItem(SHARED_PREF_USER_ID);
+    if (
+      !sourceUserId ||
+      sourceUserId === '' ||
+      sourceUserId === destinationUserId
+    ) {
+      console.log('sourceUserId is null or same as destinationUserId');
+      return;
+    }
+    baseIterableRequest<IterableResponse>({
+      method: 'GET',
+      url: ENDPOINT_GET_USER_BY_USERID,
+      data: { userId: destinationUserId },
+      validation: {}
+    })
+      .then((response) => {
+        const userData: any = response.data;
+        if (userData) {
+          const dataObj = JSON.parse(userData);
+          if (dataObj.user) {
+            this.callMergeApi(
+              '',
+              sourceUserId,
+              localStorage.getItem(SHARED_PREF_EMAIL) || '',
+              destinationUserId
+            );
+          }
+        }
+      })
```

Comment:
do we not need to handle for the case where userData does not exist?

Reply:
No. In the requirement that if API provides success response then and only then we have to do further operations.